### PR TITLE
[receiver/awscloudwatchmetrics] [2/2] final implementation of receiver

### LINKED ISSUE #15667: New component: AWS CloudWatch metrics receiver
### The purpose and use-cases of the new component

Collect metrics from AWS Cloudwatch metrics and parse them in to OTel pmetric. Some metrics on AWS, such as Lambda memory usage etc, are  only available from CloudWatch — users may want them to be put in one platform for analysis such as Prometheus with Grafana.

### Example configuration for the component

```
receivers:
    awscloudwatchmetrics:
        region: <aws_region>
        profile: <aws_profile> # optional if omitted will use default for current if the AWS credential environment variables are not set
        polling_interval: 60
        metrics:
          named: 
             - namespace: AWS/ALB
                metric_name: RequestCount
             - namespace: AWS/ALB
                metric_name: 500Count
```

### Telemetry data types supported

Metrics

### Is this a vendor-specific component?

- [X] This is a vendor-specific component
- [ ] If this is a vendor-specific component, I am proposing to contribute this as a representative of the vendor.

### Sponsor (optional)

_No response_

### Additional context

Functionality is similar to AWSContainerMetrics receiver, maybe extend that instead and make it more generic?

### PR DESCRIPTION
**Pull request by lewis262626**
**Description:** 

AWS cloudwatch metrics final implementation. Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/15667

**Link to tracking Issue:** 

https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/15667

**Testing:** 

Unit and integration testing

**Documentation:** 

README.md

### CODE REVIEW

**Comment by jpkrohling** [comment]
Please ping us when this is ready to be reviewed.

**Comment by jpkrohling** [comment]
@lewis262626, is this ready for review?

**Comment by lewis262626** [comment]
> @lewis262626, is this ready for review?

I've been meaning to finish this off, but I'm on holiday without access to my laptop. 

Should be ready by early August. There are a few bugs from what I can remember

**Comment by lewis262626** [comment]
> Hey guys, is this completed and merged ?
> @jpkrohling

Still need to finish off final testing, apologies

**Comment by ceevaaa** [comment]
> > Hey guys, is this completed and merged ?
> > @jpkrohling
> 
> Still need to finish off final testing, apologies

No worries. Thanks for all the efforts brother <3.

**Comment by michael-burt** [comment]
Hi @lewis262626 is this PR still being worked on?

**Comment by michael-burt** [comment]
Hi @lewis262626 is this PR still being worked on?

File: receiver/awscloudwatchmetricsreceiver/receiver.go [hunk 1]
```diff
@@ -56,12 +165,157 @@ func (m *metricReceiver) Shutdown(_ context.Context) error {
...
+	return pdm
+}
+
+func (m *metricReceiver) autoDiscoverRequests(ctx context.Context, auto *AutoDiscoverConfig) ([]request, error) {
+	m.logger.Debug("discovering metrics", zap.String("namespace", auto.Namespace))
+
+	requests := []request{}
+	input := &cloudwatch.ListMetricsInput{
+		Namespace: aws.String(auto.Namespace),
+	}
```
[the remark this answers is not in the document]
Comment (lewis262626) [reply receiver/awscloudwatchmetricsreceiver/receiver.go:286]: In a separate PR, yes.

**Comment by lewis262626** [comment]
> Hi @lewis262626 is this PR still being worked on?

Yes, just didn't have any motivation to finish this off. I just have the integration tests to write, then should be done

**Comment by ceevaaa** [comment]
@lewis262626 how much work is left ?

**Comment by JonathanPartain** [comment]
@lewis262626 Any plans on continuing this work? I am quite interested in this, so I'll gladly help if you want, despite my limited knowledge of go :smile:

**Comment by jaslkaur29** [comment]
Hi, does anyone know when it will be available for use?

**Pull request closed without merging**